pym/portage/dbapi/bintree.py

```diff
@@ -373,6 +380,16 @@ def __init__(self, _unused=DeprecationWarning, pkgdir=None,
 				self._pkgindex_inherited_keys,
 				chain(*self._pkgindex_translated_keys)
 			))
+			try:
+				binpkg_compression = settings.get("BINPKG_COMPRESSION", "bzip2")
+				self.compression = _compressors[binpkg_compression]
+			except KeyError as e:
+				raise ValueError("Invalid or unsupported compression method: %s" % e.args[0])
+
+			compression_binary = varexpand(shlex_split(self.compression["compress"])[0], mydict=settings)
+			if find_binary(compression_binary) is None:
+				missing_package = self.compression["package"]
+				raise ValueError("File compression unsupported %s. Missing package: %s" % (binpkg_compression, missing_package))
```

Comment:
So how about a BinpkgCompError, raise it here and catch it in emerge and quickpkg when instantiating the binarytree class?